server/main.cpp

```diff
@@ -73,42 +82,54 @@ int main() {
 			[&] () {
 				while (!closing) {
 					next_time = std::chrono::high_resolution_clock::now();
+					uint64_t current_timestamp = std::chrono::duration_cast<std::chrono::milliseconds>(next_time.time_since_epoch()).count();
 					elapsed_seconds = next_time - last_time;
 					last_time = next_time;
-					delta_accumulator += elapsed_seconds.count();
-					if (delta_accumulator >= tec::UPDATE_RATE) {
-						current_state_id++;
+					double delta = elapsed_seconds.count();
+					delta_accumulator += delta;
+					step_accumulator += delta;
+
+					if (step_accumulator >= SERVER_SIMULATE_RATE) {
+						step_accumulator -= SERVER_SIMULATE_RATE;
 						game_state_queue.ProcessEventQueue();
-						tec::GameState full_state = simulation.Simulate(tec::UPDATE_RATE, game_state_queue.GetBaseState());
-						full_state.state_id = current_state_id;
-						tec::proto::GameStateUpdate full_state_update;
-						full_state_update.set_command_id(current_state_id);
-						full_state.Out(&full_state_update);
-						tec::networking::ServerMessage full_state_update_message;
-						full_state_update_message.SetStateID(current_state_id);
-						full_state_update_message.SetMessageType(tec::networking::MessageType::GAME_STATE_UPDATE);
-						full_state_update_message.SetBodyLength(full_state_update.ByteSizeLong());
-						full_state_update.SerializeToArray(full_state_update_message.GetBodyPTR(), static_cast<int>(full_state_update_message.GetBodyLength()));
-						full_state_update_message.encode_header();
-						
-						{
-							std::lock_guard lg(server.client_list_mutex);
-							for (std::shared_ptr<tec::networking::ClientConnection> client : server.GetClients()) {
-								client->UpdateGameState(full_state);
-								if (current_state_id - client->GetLastConfirmedStateID() > tec::TICKS_PER_SECOND * 2.0) {
-									server.Deliver(client, full_state_update_message);
-									std::cout << "sending full state " << current_state_id << " to: " << client->GetID() << " client state ID was: " << client->GetLastConfirmedStateID() << std::endl;
-								}
-								else {
-									server.Deliver(client, client->PrepareGameStateUpdateMessage(current_state_id));
+						tec::GameState full_state = simulation.Simulate(SERVER_SIMULATE_RATE, game_state_queue.GetBaseState());
```

Comment:
I changed the server to use a fixed update rate, but I had to move things into an "if" block as result.
Depending how loaded the server is, this might make it spend more time in thread sleep.